x/ibc/handler_test.go

```diff
@@ -326,8 +329,11 @@ func (suite *HandlerTestSuite) TestHandleTimeoutPacket() {
 			clientA, clientB, connA, connB := suite.coordinator.SetupClientConnections(suite.chainA, suite.chainB, clientexported.Tendermint)
 			channelA, channelB := suite.coordinator.CreateChannel(suite.chainA, suite.chainB, connA, connB, channeltypes.ORDERED)
 			packet = channeltypes.NewPacket(suite.chainA.GetPacketData(suite.chainB), 1, channelA.PortID, channelA.ID, channelB.PortID, channelB.ID, uint64(suite.chainB.GetContext().BlockHeight()), uint64(suite.chainB.GetContext().BlockTime().UnixNano()))
-			err := suite.coordinator.SendPacket(suite.chainA, suite.chainB, packet, clientB)
-			suite.Require().NoError(err)
+
+			// send from chainA to chainB
+			msg := ibctransfertypes.NewMsgTransfer(channelA.PortID, channelA.ID, coin, suite.chainA.SenderAccount.GetAddress(), suite.chainB.SenderAccount.GetAddress().String(), packet.GetTimeoutHeight(), packet.GetTimeoutTimestamp())
```

Comment:
Since this is testing ibc core, this msg should be abstracted to:

```go
// Creates a transfer packet with a default token for testing purposes
ibctesting.NewMsgPacket(channelA.PortID, channelA.ID, suite.chainA.SenderAccount.GetAddress(), suite.chainB.SenderAccount.GetAddress().String(), packet.GetTimeoutHeight(), packet.GetTimeoutTimestamp())
```

Note: coin is not an argument to the function, it will just create a default coin in the packet.
That way the specific packet type used for ibc testing can be easily swapped out to a different application if necessary

Reply:
I don't understand this reasoning. These tests only refer to transfer to satisfy the handler callbacks. No application should run tests with a static result for the callbacks. This way of testing should only be used in these specific handler tests.

I added `TestCoin` to ibctesting so that the coin used in these tests are static, but I don't see any benefit to abstracting out another function in ibctesting. In general, we should aim to have ibctesting be as least application module dependent as possible. It should be a testing framework for dealing with common core ibc testing steps (setting up connection, channels, relaying packets, creating/updating clients)